rllib/algorithms/algorithm.py

```diff
@@ -1316,12 +1316,27 @@ def restore_workers(self, workers: WorkerSet):
         restored = workers.probe_unhealthy_workers()
 
         if restored:
-            from_worker = workers.local_worker() or self.workers.local_worker()
-            state = ray.put(from_worker.get_state())
+            # Figure out whether we are restoring a worker from the eval worker set.
```

Comment:
Main logic in this PR:
* Figure out whether `workers` is *the* eval worker set.
* If yes, we need to adjust the `state` of the local worker to contain the correct (eval) policy_mapping_fn and policy_to_train fn.